roaring/roaring.go

```diff
@@ -3333,9 +3375,16 @@ func popcount(x uint64) uint64 {
 }
 
 func popcountAndSlice(s, m []uint64) uint64 {
```

Comment:
let's add a comment that this *needs* to be called with slices which are exactly `bitmapN` long. I also generally agree with @alanbernstein that we need to call out the bounds checking code as such - it's likely that we'll be able to remove some or all of it as the compiler improves, so we want to be clear on why it's there in the first place.